tests/integration/create_token.py

```diff
@@ -125,7 +125,11 @@ def _get_payload_with_params(
         payload_vars["response_expires_at"] = get_response_expires_at()
 
         for key, value in extra_payload.items():
-            payload_vars[key] = value
+            if payload_vars.get("survey_metadata"):
```

Comment:
Possibly move this condtional from the loop so it's only checked once and have two loops on `if` and `else`? As in:
```python
if survey_metadata := payload_vars.get("survey_metadata").get("data"):
    for key, value:
        survey_metada[key] = value
else:
    for key, value:
         payload_vars[key] = value
```
Otherwise you create the survey metadata every single time? Also what happens to "data" object? Is it supposed to stay in payload_vars["survey_metadata"]?

Reply:
You can simplify it by doing:
```python
if survey_metadata := payload_vars.get("survey_metadata").get("data"):
    populate_with_extra_payload_items(extra_payload, survey_metadata)
else:
    populate_with_extra_payload_items(extra_payload, payload_vars)
```

Reply:
Updated 👍